File: arena-backend/src/entities/contact.entity.ts
import {
  CreateDateColumn,
  DeleteDateColumn,
  Entity,
  JoinColumn,
  ManyToOne,
  PrimaryColumn,
  UpdateDateColumn,
} from "typeorm";
import { UserEntity } from "./user.entity";

@Entity({ name: "contacts" })
export class ContactEntity {
  @PrimaryColumn({ type: "text" })
  ownerId: string;

  @PrimaryColumn({ type: "text" })
  userId: string;

  @ManyToOne(() => UserEntity)
  @JoinColumn({ name: "ownerId", referencedColumnName: "userId" })
  owner: UserEntity;

  @ManyToOne(() => UserEntity)
  @JoinColumn({ name: "userId", referencedColumnName: "userId" })
  user: UserEntity;

  @CreateDateColumn({ type: "timestamptz" })
  createdAt: Date;

  @UpdateDateColumn({ type: "timestamptz" })
  updatedAt: Date;

  @DeleteDateColumn({ type: "timestamptz", nullable: true })
  deletedAt: Date | null;
}

File: legacy/backend/src/main/java/app/sandori/arena/api/domain/contact/ContactEntity.java
package app.sandori.arena.api.domain.contact;

import app.sandori.arena.api.domain.user.UserEntity;
import app.sandori.arena.api.global.entity.BaseTimeEntity;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.Id;
import jakarta.persistence.IdClass;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "contacts")
@IdClass(ContactId.class)
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class ContactEntity extends BaseTimeEntity {

    @Id
    private String ownerId;

    @Id
    private String userId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "ownerId", referencedColumnName = "userId", insertable = false, updatable = false)
    private UserEntity owner;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "userId", referencedColumnName = "userId", insertable = false, updatable = false)
    private UserEntity user;

    public ContactEntity(String ownerId, String userId) {
        this.ownerId = ownerId;
        this.userId = userId;
    }
}
